src/cloud-api-adaptor/test/provisioner/gcp/cluster.go

```diff
@@ -0,0 +1,346 @@
+	yamlPath := "../provisioner/gcp/containerdDaemonSet.yaml"
+	err = g.DeployDaemonSet(ctx, yamlPath)
+	if err != nil {
+		return fmt.Errorf("GKE: Error injecting DaemonSet to update containerd: %v", err)
+	}
+	err = g.ApplyNodeLabels(ctx)
+	if err != nil {
+		return fmt.Errorf("GKE: Error applying node labels: %v", err)
+	}
+	return nil
+}
+
+// DeleteCluster deletes the GKE cluster
+func (g *GKECluster) DeleteCluster(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Hour)
+	defer cancel()
+
+	srv, err := container.NewService(
+		ctx, option.WithCredentialsFile(g.credentials),
+	)
+	if err != nil {
+		return fmt.Errorf("GKE: container.NewService: %v", err)
+	}
+
+	op, err := srv.Projects.Zones.Clusters.Delete(
+		g.ProjectID, g.Zone, g.clusterName,
+	).Context(ctx).Do()
+	if err != nil {
+		return fmt.Errorf("GKE: Projects.Zones.Clusters.Delete: %v", err)
+	}
+
+	log.Infof("GKE: Cluster deletion operation: %v\n", op.Name)
+
+	// Wait for the cluster to be deleted
+	activationTimeout := 30 * time.Minute
+	err = g.WaitForClusterDeleted(ctx, activationTimeout)
+	if err != nil {
+		return fmt.Errorf("GKE: error waiting for cluster to be deleted: %v", err)
+	}
+	return nil
+}
+
+// DeployDaemonSet is used here because we need to patch containerd config file.
+func (g *GKECluster) DeployDaemonSet(ctx context.Context, yamlPath string) error {
+	kubeconfigPath, err := g.GetKubeconfigFile(ctx)
+	if err != nil {
+		return err
+	}
+
+	kubeConfig, err := os.ReadFile(filepath.Clean(kubeconfigPath))
+	if err != nil {
+		return fmt.Errorf("failed to read kubeconfig file: %w", err)
+	}
+
+	config, err := clientcmd.RESTConfigFromKubeConfig(kubeConfig)
+	if err != nil {
+		return fmt.Errorf("failed to load kubeconfig: %w", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create Kubernetes client: %w", err)
+	}
+
+	yamlFile, err := os.ReadFile(filepath.Clean(yamlPath))
+	if err != nil {
+		return fmt.Errorf("failed to read DaemonSet YAML file: %w", err)
+	}
+
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(yamlFile), 4096)
+	ds := &appsv1.DaemonSet{}
+	if err := decoder.Decode(ds); err != nil {
+		return fmt.Errorf("failed to decode DaemonSet YAML: %w", err)
+	}
+
+	_, err = clientset.AppsV1().DaemonSets(ds.Namespace).Create(ctx, ds, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to deploy DaemonSet: %w", err)
+	}
+
+	log.Info("DaemonSet deployed successfully!")
+	return nil
+}
+
+// GetKubeconfigFile retrieves the path to the kubeconfig file
+func (g *GKECluster) GetKubeconfigFile(ctx context.Context) (string, error) {
+	cmd := exec.CommandContext(ctx, "gcloud", "container", "clusters", "get-credentials", g.clusterName, "--zone", g.Zone, "--project", g.ProjectID)
+	output, err := cmd.CombinedOutput()
+
+	if err != nil {
+		return "", fmt.Errorf("Failed to get cluster credentials: %v\nOutput: %s", err, output)
+	}
+
+	if g.cluster == nil {
```

Comment:
Hi @beraldoleal !

Shouldn't this check be done before exec'ing `gcloud` (first line)?